include/ebpf_program_types.h

```diff
@@ -4,48 +4,51 @@
 
 #include "ebpf_base.h"
 #include "ebpf_result.h"
-
-#include <guiddef.h>
-#if !defined(NO_CRT) && !defined(_NO_CRT_STDIO_INLINE)
-#include <stdint.h>
-#else
-typedef unsigned char uint8_t;
-typedef unsigned int uint32_t;
-typedef unsigned long long uint64_t;
-typedef unsigned short wchar_t;
-#endif
+#include "ebpf_windows.h"
 
 #define EBPF_MAX_PROGRAM_DESCRIPTOR_NAME_LENGTH 256
 #define EBPF_MAX_HELPER_FUNCTION_NAME_LENGTH 256
 
+// This is the type definition for the eBPF program type descriptor
+// when version is EBPF_PROGRAM_TYPE_DESCRIPTOR_VERSION_0.
 typedef struct _ebpf_program_type_descriptor
 {
+    ebpf_extension_header_t header;
     const char* name;
     const ebpf_context_descriptor_t* context_descriptor;
     GUID program_type;
     uint32_t bpf_prog_type;
     char is_privileged;
 } ebpf_program_type_descriptor_t;
```

Comment:
nit: should we instead define this struct as `ebpf_program_type_descriptor_t_0` or `ebpf_program_type_descriptor_t_version_0` and then:

```
#define ebpf_program_type_descriptor_t ebpf_program_type_descriptor_t_version_0
```
Whenever there is a version bump in future, we will need to define a `version_1` struct and also update the above `#define`?

Reply:
That sounds like a fine suggestion to me.

Reply:
There were two options:
1) Change the type identifier when version changed and support older versions.
2) Version numbers need not be changed, even if data structure changes, as long as they are backward compatible. Version changes will be made only when there is a breaking change, and there will be a corresponding major version change in eBPF as well, at which point older extensions/clients will no longer work.

We chose the second option (which is the style used for NDIS structs) - where the type identifier does not change even when struct definition changes maintaining backward compatibility. A version change indicates breaking change and extensions/clients need to be re-compiled.